src/main/java/com/coveros/selenified/utilities/Hub.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 Coveros, Inc.
+ *
+ * This file is part of Selenified.
+ *
+ * Selenified is licensed under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.coveros.selenified.utilities;
+
+import com.coveros.selenified.exceptions.InvalidHubException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static com.coveros.selenified.utilities.Property.HUB;
+import static com.coveros.selenified.utilities.Property.getProgramProperty;
+
+/**
+ * Utilities class to interact with some remote hub driver, retrieving information such as username, key, and test status
+ *
+ * @author Max Saperstone
+ * @version 3.2.1
+ * @lastupdate 8/23/2019
+ */
+public class Hub {
+
+    private static final String HUB_ISN_T_SET = "Hub isn't set";
+    private static final String HUB_USER = "HUB_USER";
+    private static final String HUB_PASS = "HUB_PASS";
+    private URL hubURL;
+    private String username = null;
+    private String password = null;
+
+    public Hub() throws MalformedURLException {
+        if (!isHubSet()) {
+            throw new InvalidHubException(HUB_ISN_T_SET);
+        }
+        String hubProperty = getProgramProperty(HUB);
+        try {
+            this.hubURL = new URL(hubProperty);
+        } catch (MalformedURLException e) {
+            throw new InvalidHubException("Hub '" + hubProperty + "' isn't valid. Must contain protocol, optionally credentials, and endpoint");
+        }
+        setUserInfo();
+        String credentials = username == null ? "" : username + ":" + password + "@";
+        String port = hubURL.getPort() == -1 ? "" : ":" + String.valueOf(hubURL.getPort());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Directly append the argument of String.valueOf(). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1153)